### PR TITLE
Added hahsnode_script

### LINKED ISSUE #738: Script to display top posts from Hashnode
# Description

Add a Script to display top posts from Hashnode. Take a look at the Hashnode API [here](https://engineering.hashnode.com/introducing-hashnode-graphql-api-public-beta-cjydzvp59001q2gs1b5zxaeaf) and the Playground [here](https://api.hashnode.com/). 

The Script should ideally fetch the top posts that are trending on Hashnode as of now. You can also use flags to pass arguments for various posts trending in specific topics on Hashnode.

## Language

- [X] Go
- [X] Javascript
- [X] Python

## Checklist:

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| Hashnode Top Posts Script | Fetch top posts trending on Hashnode with Hashnode API | MWOC, Python, Javascript, Golang | |

**Is your feature request related to a problem? Please describe.**: A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

**Describe the solution you'd like**: A clear and concise description of what you want to happen.

**Additional context**: Add any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by Sukriti-sood**
Fixes #738 

This Script will Will Store Best required Posts from Hashnode With their title , Brief Summary and author name in a CSV FIle.

Sample Output is Shown in README.

### CODE REVIEW

**Comment by Sukriti-sood** [comment]
@HarshCasper @vybhav72954 Please review this PR.

File: JavaScript/Hashnode_script/Readme.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Hashnode Best Posts Script
+
+A Script to scrape reviews from Amazon.
+
+## Setup
+- Clone the folder.  
+- Open command line and change the directory to your folder by using ```cd foldername```.
+3. Run the command ```npm install```.
```
Comment (vybhav72954) [inline JavaScript/Hashnode_script/Readme.md:8]: Use bullets only please XD

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by vybhav72954** [comment]
@Sukriti-sood Can you rename your directory `Hashnode_Scripts`

File: JavaScript/Hashnode_Scripts/Readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Hashnode Best Posts Script
+
+Script to store Best Posts required from Hashnode. 
+
+## Setup
+- Clone the folder.  
+- Open command line and change the directory to your folder by using ```cd foldername```.
+- Run the command ```npm install```.
+- Then run the script using command ```node script.js no_of_posts```, Like ```node script.js 14```
```
Comment (HarshCasper) [inline JavaScript/Hashnode_Scripts/Readme.md:9]: Can you document this in a better way?

File: JavaScript/Hashnode_Scripts/Readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Hashnode Best Posts Script
+
+Script to store Best Posts required from Hashnode.
```
Comment (HarshCasper) [inline JavaScript/Hashnode_Scripts/Readme.md:3]: Add a bit more descriptive outline

File: JavaScript/Hashnode_Scripts/script.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        }
+    }
+
+    const csv=new ObjectsToCsv(posts_data)
+
+    await csv.toDisk("./output.csv")
+
+}
+
+fetchPosts(post_count);
```
Comment (HarshCasper) [inline JavaScript/Hashnode_Scripts/script.js:71]: Add a new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: JavaScript/Hashnode_Scripts/script.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+// all requires
+const fetch = require("node-fetch");
+const ObjectsToCsv=require("objects-to-csv");
+
+
+//variables required
+var page=0;
```
Comment (sohamsshah) [inline JavaScript/Hashnode_Scripts/script.js:7]: `let` and `const` is a good JS practice. Replace all `var` with `let` for better code readability and flexibility.

File: JavaScript/Hashnode_Scripts/script.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+var post_count=process.argv[2];
+
+
```
Comment (sohamsshah) [inline JavaScript/Hashnode_Scripts/script.js:10]: Remove extra lines. Use a linter or similar for that

File: JavaScript/Hashnode_Scripts/script.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        var posts=ApiResponse.data.storiesFeed;
+
+        current_post_count+=posts.length;
+
+        page++;
+
+        for(var post of posts){
+            count++;
+
+            var post_object={}
```
Comment (sohamsshah) [inline JavaScript/Hashnode_Scripts/script.js:50]: In JavaScript, generally the variable conventions generally followed are TitleCase or camelCase. Replace all the Underscore_case variable names with appropriate ones. Just a small enhancement, pardon for being too picky haha with best practices!

**Review by sohamsshah — CHANGES_REQUESTED** [review]
Functionality-wise the script LGTM.
Revert back with the suggested enhancements and the PR is good to merge!

File: JavaScript/Hashnode_Scripts/script.js (hunk 7 above)
Comment (Sukriti-sood) [reply JavaScript/Hashnode_Scripts/script.js]: Changes Done, Please Have a Look.

File: JavaScript/Hashnode_Scripts/script.js (hunk 4 above)
Comment (Sukriti-sood) [reply JavaScript/Hashnode_Scripts/script.js]: Changes Done, Please Have a Look.

**Comment by vybhav72954** [comment]
Hey, @Sukriti-sood Not sure how and why this one happened. But your PR is affecting other files as well. Am sorry but I think you will have to make a New PR though. No swat, just a new branch from master, commit your latest files, and I will get to them asap. Thank you.

**Comment by vybhav72954** [comment]
Migrated to #983

**Pull request closed without merging**